modules/users/server/controllers/users/users.profile.server.controller.js

```diff
@@ -60,39 +60,83 @@ exports.changeProfilePicture = function (req, res) {
   var user = req.user;
   var upload = multer(config.uploads.profileUpload).single('newProfilePicture');
   var profileUploadFileFilter = require(path.resolve('./config/lib/multer')).profileUploadFileFilter;
+  var tmpImageUrl;
 
   // Filtering to upload only images
   upload.fileFilter = profileUploadFileFilter;
-
+  
   if (user) {
-    upload(req, res, function (uploadError) {
-      if (uploadError) {
-        return res.status(400).send({
-          message: 'Error occurred while uploading profile picture'
-        });
-      } else {
-        user.profileImageURL = config.uploads.profileUpload.dest + req.file.filename;
+    tmpImageUrl = user.profileImageURL;
+    uploadImage()
+      .then(updateUser)
+      .then(deleteOldImage)
+      .then(login)
+      .then(function () {
+        res.json(user);
```

Comment:
Will the user sent back with the response here be the updated user (after the update method has completed)? Even though `user` is defined within the same scope, and is available in the `updateUser` closure, it's never updated; see my comment below in the `updateUser` method.